## Programa Ingenias+ Data Science

In [ ]:
  #from google.colab import drive
  #drive.mount('/content/drive')

In [5]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# Sección nueva

## Trabajando con Pandas

1) Importa las librerias requeridas: `Pandas` y `NumPy`.  

Una buena practica a la hora de escribir codigo es de comentar siempre que estoy haciendo (usando `#`) en la linea anterior. Por mas obvio que parezca lo que voy a hacer, puede ser que tenga que compartir el documento con otras personas y no sepan que quise hacer en cada linea. O que yo mismo lea el codigo tiempo despues y no recuerde porque lo hice.

In [6]:
import pandas as pd
import numpy as np

2) Lee el archivo csv `life_expectancy_data` usando `pandas`. Guardalo en una variable llamada `expvida`.

In [8]:
  # Leo los datos en un DataFrame
expvida = pd.read_csv('/content/drive/MyDrive/Life_Expectancy_Data.csv')


In [54]:
expvida

Country  Year      Status  life_expectancy  Adult Mortality  \
0     Afghanistan  2015  Developing             65.0            263.0   
1     Afghanistan  2014  Developing             59.9            271.0   
2     Afghanistan  2013  Developing             59.9            268.0   
3     Afghanistan  2012  Developing             59.5            272.0   
4     Afghanistan  2011  Developing             59.2            275.0   
...           ...   ...         ...              ...              ...   
2933     Zimbabwe  2004  Developing             44.3            723.0   
2934     Zimbabwe  2003  Developing             44.5            715.0   
2935     Zimbabwe  2002  Developing             44.8             73.0   
2936     Zimbabwe  2001  Developing             45.3            686.0   
2937     Zimbabwe  2000  Developing             46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  measles  \
0                62     0.01               71.279624         65.0     1154   
1                64     0.01               73.523582         62.0      492   
2                66     0.01               73.219243         64.0      430   
3                69     0.01               78.184215         67.0     2787   
4                71     0.01                7.097109         68.0     3013   
...             ...      ...                     ...          ...      ...   
2933             27     4.36                0.000000         68.0       31   
2934             26     4.06                0.000000          7.0      998   
2935             25     4.43                0.000000         73.0      304   
2936             25     1.72                0.000000         76.0      529   
2937             24     1.68                0.000000         79.0     1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
0     ...    6.0               8.16         65.0        0.1  584.259210   
1     ...   58.0               8.18         62.0        0.1  612.696514   
2     ...   62.0               8.13         64.0        0.1  631.744976   
3     ...   67.0               8.52         67.0        0.1  669.959000   
4     ...   68.0               7.87         68.0        0.1   63.537231   
...   ...    ...                ...          ...        ...         ...   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
0     33736494.0                   17.2                 17.3   
1       327582.0                   17.5                 17.5   
2     31731688.0                   17.7                 17.7   
3      3696958.0                   17.9                 18.0   
4      2978599.0                   18.2                 18.2   
...          ...                    ...                  ...   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427       10.0  
2936            

[texto del vínculo](https:// [texto del vínculo](https://))3) ¿Que tipo de estructura de datos contiene la variable `expvida`?

In [10]:
# Obtengo el tipo de estructura de datos
type(expvida)

pandas.core.frame.DataFrame

4) ¿Cuantas filas y columnas tiene `expvida`?

In [11]:
# Uso .shape para obtener cantidad de filas y columnas
expvida.shape


(2938, 22)

**Numero de filas**:

2938

**Numero de columnas**:

22

5) ¿Cual es el nombre de las columnas contenidas en `expvida`?

In [12]:
# Obtengo el nombre de las columnas
expvida.columns

Index(['Country', 'Year', 'Status', 'Life expectancy ', 'Adult Mortality',
       'infant deaths', 'Alcohol', 'percentage expenditure', 'Hepatitis B',
       'Measles ', ' BMI ', 'under-five deaths ', 'Polio', 'Total expenditure',
       'Diphtheria ', ' HIV/AIDS', 'GDP', 'Population',
       ' thinness  1-19 years', ' thinness 5-9 years',
       'Income composition of resources', 'Schooling'],
      dtype='object')

6) Inspecciona las primeras 10 filas de `expvida`

In [13]:
# Inspecciono las primeras filas
expvida.head(10)

Country  Year      Status  Life expectancy   Adult Mortality  \
0  Afghanistan  2015  Developing              65.0            263.0   
1  Afghanistan  2014  Developing              59.9            271.0   
2  Afghanistan  2013  Developing              59.9            268.0   
3  Afghanistan  2012  Developing              59.5            272.0   
4  Afghanistan  2011  Developing              59.2            275.0   
5  Afghanistan  2010  Developing              58.8            279.0   
6  Afghanistan  2009  Developing              58.6            281.0   
7  Afghanistan  2008  Developing              58.1            287.0   
8  Afghanistan  2007  Developing              57.5            295.0   
9  Afghanistan  2006  Developing              57.3            295.0   

   infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   ...  \
0             62     0.01               71.279624         65.0      1154  ...   
1             64     0.01               73.523582         62.0       492  ...   
2             66     0.01               73.219243         64.0       430  ...   
3             69     0.01               78.184215         67.0      2787  ...   
4             71     0.01                7.097109         68.0      3013  ...   
5             74     0.01               79.679367         66.0      1989  ...   
6             77     0.01               56.762217         63.0      2861  ...   
7             80     0.03               25.873925         64.0      1599  ...   
8             82     0.02               10.910156         63.0      1141  ...   
9             84     0.03               17.171518         64.0      1990  ...   

   Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  Population  \
0    6.0               8.16         65.0        0.1  584.259210  33736494.0   
1   58.0               8.18         62.0        0.1  612.696514    327582.0   
2   62.0               8.13         64.0        0.1  631.744976  31731688.0   
3   67.0               8.52         67.0        0.1  669.959000   3696958.0   
4   68.0               7.87         68.0        0.1   63.537231   2978599.0   
5   66.0               9.20         66.0        0.1  553.328940   2883167.0   
6   63.0               9.42         63.0        0.1  445.893298    284331.0   
7   64.0               8.33         64.0        0.1  373.361116   2729431.0   
8   63.0               6.73         63.0        0.1  369.835796  26616792.0   
9   58.0               7.43         58.0        0.1  272.563770   2589345.0   

    thinness  1-19 years   thinness 5-9 years  \
0                   17.2                 17.3   
1                   17.5                 17.5   
2                   17.7                 17.7   
3                   17.9                 18.0   
4                   18.2                 18.2   
5                   18.4                 18.4   
6                   18.6                 18.7   
7                   18.8                 18.9   
8                   19.0                 19.1   
9                   19.2                 19.3   

   Income composition of resources  Schooling  
0                            0.479       10.1  
1                            0.476       10.0  
2                            0.470        9.9  
3                            0.463        9.8  
4                            0.454        9.5  
5                            0.448        9.2  
6                            0.434        8.9  
7                            0.433        8.7  
8                            0.415        8.4  
9                            0.405        8.1  

[10 rows x 22 columns]

7) Y ahora las 10 ultimas.

In [14]:
# Inspecciono las ultimas filas
expvida.tail(10)

Country  Year      Status  Life expectancy   Adult Mortality  \
2928  Zimbabwe  2009  Developing              50.0            587.0   
2929  Zimbabwe  2008  Developing              48.2            632.0   
2930  Zimbabwe  2007  Developing              46.6             67.0   
2931  Zimbabwe  2006  Developing              45.4              7.0   
2932  Zimbabwe  2005  Developing              44.6            717.0   
2933  Zimbabwe  2004  Developing              44.3            723.0   
2934  Zimbabwe  2003  Developing              44.5            715.0   
2935  Zimbabwe  2002  Developing              44.8             73.0   
2936  Zimbabwe  2001  Developing              45.3            686.0   
2937  Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
2928             30     4.64                1.040021         73.0       853   
2929             30     3.56               20.843429         75.0         0   
2930             29     3.88               29.814566         72.0       242   
2931             28     4.57               34.262169         68.0       212   
2932             28     4.14                8.717409         65.0       420   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
2928  ...   69.0               6.26         73.0       18.1   65.824121   
2929  ...   75.0               4.96         75.0       20.5  325.678573   
2930  ...   73.0               4.47         73.0       23.7  396.998217   
2931  ...   71.0               5.12          7.0       26.8  414.796232   
2932  ...   69.0               6.44         68.0       30.3  444.765750   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
2928   1381599.0                    7.5                  7.4   
2929  13558469.0                    7.8                  7.8   
2930   1332999.0                    8.2                  8.2   
2931  13124267.0                    8.6                  8.6   
2932    129432.0                    9.0                  9.0   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
2928                            0.419        9.9  
2929                            0.421        9.7  
2930                            0.414        9.6  
2931                            0.408        9.5  
2932                            0.406        9.3  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427       10.0  
2936                            0.427        9.8  
2937                            0.434        9.8  

[10 rows x 22 columns]

8) ¿Que tipos de datos contiene cada una de las columnas de `expvida`?

In [15]:
# Tipo de datos en cada columna
expvida.dtypes

Country                             object
Year                                 int64
Status                              object
Life expectancy                    float64
Adult Mortality                    float64
infant deaths                        int64
Alcohol                            float64
percentage expenditure             float64
Hepatitis B                        float64
Measles                              int64
 BMI                               float64
under-five deaths                    int64
Polio                              float64
Total expenditure                  float64
Diphtheria                         float64
 HIV/AIDS                          float64
GDP                                float64
Population                         float64
 thinness  1-19 years              float64
 thinness 5-9 years                float64
Income composition of resources    float64
Schooling                          float64
dtype: object

9) ¿Hay valores faltantes en alguna de las columnas de `expvida`?

In [16]:
# Obtengo la cantidad de datos faltantes en cada columna
datos_faltantes = expvida.isnull().sum()
print(datos_faltantes)

Country                              0
Year                                 0
Status                               0
Life expectancy                     10
Adult Mortality                     10
infant deaths                        0
Alcohol                            194
percentage expenditure               0
Hepatitis B                        553
Measles                              0
 BMI                                34
under-five deaths                    0
Polio                               19
Total expenditure                  226
Diphtheria                          19
 HIV/AIDS                            0
GDP                                448
Population                         652
 thinness  1-19 years               34
 thinness 5-9 years                 34
Income composition of resources    167
Schooling                          163
dtype: int64


10) Selecciona el maximo de datos faltantes. Calcula que porcentaje de los datos representan los datos faltantes.

In [17]:
# Obtengo el maximo valor de datos faltantes y lo guardo en la variable masnan
masnan = expvida.isnull().sum().max()

print(f"El valor máximo de datos faltantes es {masnan}")

El valor máximo de datos faltantes es 652


In [18]:
# Obtengo la cantidad de filas total y divido el maximo valor de datos faltantes por este numero
# y lo multiplico por 100

filas_total= len(expvida)

resultado1 = (masnan / filas_total) * 100

print(resultado1)

22.19196732471069


In [19]:
# Imprimo el maximo porcentaje de datos faltantes. Uso f-strings.

print(f"El porcentaje máximo de filas nulas es igual a {resultado1:.2f} %")


El porcentaje máximo de filas nulas es igual a 22.19 %


In [21]:
# Calcular el porcentaje de valores nulos por columna
porcentaje_nulos = expvida.isnull().mean() * 100

# Obtener el porcentaje máximo
porcentaje = porcentaje_nulos.max()

# Convertirlo a string si quieres mostrarlo
por_str = str(round(porcentaje, 2)) + " %"

print("El porcentaje máximo de filas nulas es igual a", por_str)


El porcentaje máximo de filas nulas es igual a 22.19 %


In [22]:
# Imprimo el maximo porcentaje de datos faltantes. Uso f-strings.

print(f"El porcentaje máximo de filas nulas es igual a {resultado1:.2f} %")

# Asigno el valor resultado1 al porcentaje
porcentaje = resultado1

por_str = str(porcentaje)

type(porcentaje)

type(por_str)

print(type(por_str))

El porcentaje máximo de filas nulas es igual a 22.19 %
<class 'str'>


la variable porcentaje no esta definida

- str.format()
- Template Class
- f-strings

#### Ahora vamos a realizar algunos calculos y manipulaciones sobre las columnas.

> Agregar bloque entrecomillado



Una función útil para este fin es: `.rename()`. Y podemos utilizarla de la siguiente manera: Si `df` es mi DataFrame y quiero renombrar la columna `columna` como `columna1`, y la columna `columna2` como `columna3`, utilizo la siguiente sintaxis:
  
`df.rename(columns={"columna": "columna1", "columna2": "columna3"}, inplace=True)`

11) Renombra la columna 'Life expectancy ' como 'life_expectancy', ' BMI ' como 'bmi' y ''Measles ' como 'measles'. No olvides de usar `inplace` para modificar el DataFrame original. Luego chequea que el cambio se haya hecho correctamente.

In [24]:
# Renombro columnas del DataFrame


expvida.rename(columns={
    'Life expectancy ': 'life_expectancy',
    ' BMI ': 'bmi',
    'Measles ': 'measles'
}, inplace=True)



In [25]:
# Chequeo que se hayan reemplazado correctamente
print(expvida.columns)


Index(['Country', 'Year', 'Status', 'life_expectancy', 'Adult Mortality',
       'infant deaths', 'Alcohol', 'percentage expenditure', 'Hepatitis B',
       'measles', 'bmi', 'under-five deaths ', 'Polio', 'Total expenditure',
       'Diphtheria ', ' HIV/AIDS', 'GDP', 'Population',
       ' thinness  1-19 years', ' thinness 5-9 years',
       'Income composition of resources', 'Schooling'],
      dtype='object')


12) ¿Cual es el promedio de expectativa de vida (Life expectancy) en general?

In [26]:
# Obtengo el promedio de la columna
expvida['life_expectancy'].mean()



np.float64(69.22493169398908)

13) ¿Cual es el promedio de expectativa de vida (Life expectancy) por país?

In [27]:
# Obtengo el promedio por pais
expvida.groupby('Country')['life_expectancy'].mean()

Country
Afghanistan                           58.19375
Albania                               75.15625
Algeria                               73.61875
Angola                                49.01875
Antigua and Barbuda                   75.05625
                                        ...   
Venezuela (Bolivarian Republic of)    73.38750
Viet Nam                              74.77500
Yemen                                 63.86250
Zambia                                53.90625
Zimbabwe                              50.48750
Name: life_expectancy, Length: 193, dtype: float64

14) ¿Cual es la expectativa de vida mas alta y mas baja? Usa la funcion de pandas y la funcion de NumPy y compara el resultado. ¿Que paises tienen esas expectativas de vida?

In [ ]:
# Maxima expectativa de vida usando numpy
#maxima = np.array(expvida['life_expectancy'])
#maxim_exp = np.max(array)

In [29]:
# Expectativa de vida máxima y mínima usando pandas
max_exp_pandas = expvida['life_expectancy'].max()
min_exp_pandas = expvida['life_expectancy'].min()

# Países con la expectativa de vida más alta y más baja
pais_max = expvida.loc[expvida['life_expectancy'] == max_exp_pandas, 'Country'].values[0]
pais_min = expvida.loc[expvida['life_expectancy'] == min_exp_pandas, 'Country'].values[0]

print(f"Máxima expectativa de vida (pandas): {max_exp_pandas} años en {pais_max}")
print(f"Mínima expectativa de vida (pandas): {min_exp_pandas} años en {pais_min}")


Máxima expectativa de vida (pandas): 89.0 años en Belgium
Mínima expectativa de vida (pandas): 36.3 años en Haiti


In [31]:
import numpy as np

# Convertir la columna 'life_expectancy' a un array de NumPy
maxima = np.array(expvida['life_expectancy'])

# Máxima y mínima expectativa de vida usando NumPy
maxim_exp_numpy = np.max(maxima)
minim_exp_numpy = np.min(maxima)

# Verificar que el valor máximo y mínimo realmente exista
pais_max_numpy = expvida.loc[expvida['life_expectancy'] == maxim_exp_numpy, 'Country']
pais_min_numpy = expvida.loc[expvida['life_expectancy'] == minim_exp_numpy, 'Country']

# Asegurarnos de que los resultados no están vacíos
if not pais_max_numpy.empty:
    pais_max_numpy = pais_max_numpy.values[0]
else:
    pais_max_numpy = 'No encontrado'

if not pais_min_numpy.empty:
    pais_min_numpy = pais_min_numpy.values[0]
else:
    pais_min_numpy = 'No encontrado'

print(f"Máxima expectativa de vida (NumPy): {maxim_exp_numpy} años en {pais_max_numpy}")
print(f"Mínima expectativa de vida (NumPy): {minim_exp_numpy} años en {pais_min_numpy}")



Máxima expectativa de vida (NumPy): nan años en No encontrado
Mínima expectativa de vida (NumPy): nan años en No encontrado


In [32]:
expvida

Country  Year      Status  life_expectancy  Adult Mortality  \
0     Afghanistan  2015  Developing             65.0            263.0   
1     Afghanistan  2014  Developing             59.9            271.0   
2     Afghanistan  2013  Developing             59.9            268.0   
3     Afghanistan  2012  Developing             59.5            272.0   
4     Afghanistan  2011  Developing             59.2            275.0   
...           ...   ...         ...              ...              ...   
2933     Zimbabwe  2004  Developing             44.3            723.0   
2934     Zimbabwe  2003  Developing             44.5            715.0   
2935     Zimbabwe  2002  Developing             44.8             73.0   
2936     Zimbabwe  2001  Developing             45.3            686.0   
2937     Zimbabwe  2000  Developing             46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  measles  \
0                62     0.01               71.279624         65.0     1154   
1                64     0.01               73.523582         62.0      492   
2                66     0.01               73.219243         64.0      430   
3                69     0.01               78.184215         67.0     2787   
4                71     0.01                7.097109         68.0     3013   
...             ...      ...                     ...          ...      ...   
2933             27     4.36                0.000000         68.0       31   
2934             26     4.06                0.000000          7.0      998   
2935             25     4.43                0.000000         73.0      304   
2936             25     1.72                0.000000         76.0      529   
2937             24     1.68                0.000000         79.0     1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
0     ...    6.0               8.16         65.0        0.1  584.259210   
1     ...   58.0               8.18         62.0        0.1  612.696514   
2     ...   62.0               8.13         64.0        0.1  631.744976   
3     ...   67.0               8.52         67.0        0.1  669.959000   
4     ...   68.0               7.87         68.0        0.1   63.537231   
...   ...    ...                ...          ...        ...         ...   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
0     33736494.0                   17.2                 17.3   
1       327582.0                   17.5                 17.5   
2     31731688.0                   17.7                 17.7   
3      3696958.0                   17.9                 18.0   
4      2978599.0                   18.2                 18.2   
...          ...                    ...                  ...   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427       10.0  
2936            

In [33]:
# Maxima expectativa de vida usando pandas
maxima_exp = expvida['life_expectancy'].max()
print(maxima_exp)

89.0


In [35]:
# Minima expectativa de vida usando numpy
# Mínima expectativa de vida usando NumPy
minima = np.array(expvida['life_expectancy'])
minim_exp = np.min(minima)  # Usamos 'minima' en lugar de 'array'

print(f"La mínima expectativa de vida es: {minim_exp} años")


La mínima expectativa de vida es: nan años


In [41]:
# Minima expectativa de vida usando numpy
minima = np.array(expvida['life_expectancy'])
minim_exp = np.min(minima) # Changed 'array' to 'minima'


In [38]:
# Minima expectativa de vida usando pandas
minima_exp = expvida['life_expectancy'].min()

In [39]:
# Muestro los resultados maximos
fila_maxima = expvida.loc[expvida['Country'].idxmax()]
print(f"La fila con el valor máximo es: {fila_maxima}")

La fila con el valor máximo es: Country                              Zimbabwe
Year                                     2015
Status                             Developing
life_expectancy                          67.0
Adult Mortality                         336.0
infant deaths                              22
Alcohol                                   NaN
percentage expenditure                    0.0
Hepatitis B                              87.0
measles                                     0
bmi                                      31.8
under-five deaths                          32
Polio                                    88.0
Total expenditure                         NaN
Diphtheria                               87.0
 HIV/AIDS                                 6.2
GDP                                 118.69383
Population                         15777451.0
 thinness  1-19 years                     5.6
 thinness 5-9 years                       5.5
Income composition of resources         0.507
Sc

In [42]:
# Muestro los resultados minimos
print(minima_exp)


36.3


In [43]:
# Filtro las columnas para mostrar solo las filas que tengan valor
# igual al maximo
expvida.loc[expvida['life_expectancy'] == maxima_exp]

Country  Year      Status  life_expectancy  Adult Mortality  \
241       Belgium  2014   Developed             89.0             76.0   
915       Finland  2014  Developing             89.0             78.0   
937        France  2008  Developing             89.0             88.0   
938        France  2007  Developing             89.0             89.0   
995       Germany  2014   Developed             89.0             69.0   
1293        Italy  2004   Developed             89.0             66.0   
1850  New Zealand  2010   Developed             89.0             71.0   
1916       Norway  2009   Developed             89.0             67.0   
2056     Portugal  2014   Developed             89.0             78.0   
2433        Spain  2007   Developed             89.0             72.0   
2513       Sweden  2007   Developed             89.0             63.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  measles  \
241               0    12.60             7163.348923         98.0       70   
915               0     8.80             6164.455402          NaN        0   
937               3    11.90             7002.785925         47.0      604   
938               3    12.20               64.737149         42.0       39   
995               2    11.03              941.756291         88.0      443   
1293              2     8.98             4270.914882         96.0      599   
1850              0     9.62             6664.148708          9.0       43   
1916              0     6.68              142.374943          NaN        2   
2056              0     9.88              271.254553         98.0        0   
2433              2    11.05              510.932701         96.0      267   
2513              0     6.90             7593.391622          NaN        1   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS          GDP  \
241   ...   99.0               1.59         99.0        0.1  47439.39684   
915   ...   98.0               9.68         98.0        0.1  49914.61864   
937   ...   98.0               1.57         98.0        0.1  45413.65710   
938   ...   99.0               1.45         98.0        0.1    416.58397   
995   ...   94.0              11.30         95.0        0.1   4792.65288   
1293  ...   97.0               8.49         94.0        0.1  31174.56118   
1850  ...   93.0              11.20         93.0        0.1  33691.34837   
1916  ...   94.0               9.49         94.0        0.1    817.77681   
2056  ...   98.0               9.50         98.0        0.1   2277.53613   
2433  ...   96.0               8.36         96.0        0.1   3279.41400   
2513  ...   98.0               8.92         98.0        0.1  53324.37937   

      Population   thinness  1-19 years   thinness 5-9 years  \
241     112957.0                    1.0                  1.0   
915    5461512.0                    0.9                  0.8   
937    6437499.0                    0.6                  0.6   
938    6416229.0                    0.6                  0.6   
995      89825.0                    1.1                  1.1   
1293  57685327.0                    0.5                  0.5   
1850         NaN                    0.3                  0.3   
1916   4828726.0                    0.7                  0.7   
2056     14162.0                    0.7                  0.5   
2433   4522683.0                    0.6                  0.5   
2513    914892.0                    1.3                  1.3   

      Income composition of resources  Schooling  
241                             0.890       16.3  
915                             0.890       17.0  
937                             0.877       16.1  
938                             0.873       16.1  
995                             0.920       17.0  
1293                            0.846       15.6  
1850                            0.899       20.3  
1916                            0.936       17.4  
2056                            0.837       16.8  
2433

In [44]:
# Filtro las columnas para mostrar solo las filas que tengan valor
# igual al minimo
expvida.loc[expvida['life_expectancy'] == minima_exp]

Country  Year      Status  life_expectancy  Adult Mortality  \
1127   Haiti  2010  Developing             36.3            682.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  measles  \
1127             23     5.76               36.292918          NaN        0   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
1127  ...   66.0                8.9         66.0        1.9  662.279518   

      Population   thinness  1-19 years   thinness 5-9 years  \
1127   9999617.0                    4.0                  4.0   

      Income composition of resources  Schooling  
1127                             0.47        8.6  

[1 rows x 22 columns]

\15) ¿Cuantas categorias encontramos en la columna "Status"? _**Pista**_: Lee acerca de la función [unique()](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.unique.html?highlight=unique#pandas.Series.unique)

In [45]:
# Obtengo los valores unicos de la columna status
valores_unicos = expvida['Status'].unique()
print(valores_unicos)

['Developing' 'Developed']


16) ¿Cuantos paises se ubican en cada una de esas categorias de "Status"? _**Pista**_: Lee acerca de la función [value_counts()](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.value_counts.html)





In [46]:
# Cuento cuantos valores hay de cada categoria
expvida.groupby('Status')['Country'].value_counts()

Status      Country              
Developed   Australia                16
            Austria                  16
            Belgium                  16
            Bulgaria                 16
            Croatia                  16
                                     ..
Developing  Niue                      1
            Palau                     1
            Saint Kitts and Nevis     1
            San Marino                1
            Tuvalu                    1
Name: count, Length: 193, dtype: int64

18) ¿Cuantos paises presentaron algun caso de Sarampion (**Measles**) en el 2015? **_Pista_**: Lee acerca de la función [.count()](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.count.html?highlight=count#pandas.DataFrame.count)

In [51]:
# Filtramos el DataFrame para el año 2015
# Verificar los nombres de las columnas
print(expvida.columns)
# Filtrar el DataFrame para el año 2015
paisesSarampion = expvida[(expvida['Year'] == 2015)]

# Filtrar los países con casos de sarampión (measles)
paisesConSarampion = paisesSarampion[paisesSarampion['measles'] > 0]

# Contar el número de países con casos de sarampión
num_paises_sarampion = paisesConSarampion['Country'].nunique()

print(f"El número de países que tuvieron casos de sarampión en 2015 es: {num_paises_sarampion}")



Index(['Country', 'Year', 'Status', 'life_expectancy', 'Adult Mortality',
       'infant deaths', 'Alcohol', 'percentage expenditure', 'Hepatitis B',
       'measles', 'bmi', 'under-five deaths ', 'Polio', 'Total expenditure',
       'Diphtheria ', ' HIV/AIDS', 'GDP', 'Population',
       ' thinness  1-19 years', ' thinness 5-9 years',
       'Income composition of resources', 'Schooling'],
      dtype='object')
El número de países que tuvieron casos de sarampión en 2015 es: 121


In [52]:
# Cuento cuantas filas tienen valores distintos de cero
cantidad_filas = (expvida['measles'] != 0).sum()
print(cantidad_filas)

1955
